apps/login/src/lib/server/idp-intent.ts

```diff
@@ -230,57 +260,152 @@ export async function processIDPCallback({
       }
 
       return { error: t("errors.sessionCreationFailed") };
+    } catch (error) {
+      console.error("[IDP Process] Error linking IDP:", error);
+      const errorMessage = error instanceof Error ? error.message : t("errors.unknownError");
+      let params = buildRedirectParams({ error: errorMessage });
+      if (error && typeof error === "object" && "code" in error && (error as { code: unknown }).code === 6) {
```

Comment:
I don't think the as casting is needed but maybe it would be easier to just use 

```js
import { ConnectError, Code } from "@zitadel/client";

if(error instanceof ConnectError && error.code === Code.AlreadyExists) {
  params = buildRedirectParams({ error: "external_idp_taken" });
}
```